src/lib.rs

```diff
@@ -210,31 +229,40 @@ impl RandomXVM {
     }
 
     pub fn reinit_cache(&self, cache: &RandomXCache) {
+        //no way to check if this fails, c code does not return anything
         unsafe {
             randomx_vm_set_cache(self.vm, cache.cache);
         }
     }
 
     pub fn reinit_dataset(&self, dataset: &RandomXDataset) {
+        //no way to check if this fails, c code does not return anything
         unsafe {
             randomx_vm_set_dataset(self.vm, dataset.dataset);
         }
     }
 
-    pub fn calculate_hash(&self, input: &str) -> Vec<u8> {
+    pub fn calculate_hash(&self, input: &str) -> Result<Vec<u8>, CreationError> {
```

Comment:
`CreationError` does not sound like the correct Error type here. 

At this point, the VM has been created and we're checking hashes / mining with it

Reply:
I will recreate all the error class